.devcontainer/Dockerfile

```diff
@@ -0,0 +1,63 @@
+# Use the official Node.js image as base
+# This matches the Node version requirement from package.json (^20.19.0 || >=22.12.0)
+ARG NODE_VERSION=22
+FROM node:${NODE_VERSION}-bookworm
+
+# Install additional OS packages needed for development
+# - python3, make, g++: Required for building native node modules (like sqlite3)
+# - curl, wget: Useful for downloading files
+# - git: Already in base image but ensuring latest version
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install --no-install-recommends \
+    python3 \
+    python3-pip \
+    make \
+    g++ \
+    build-essential \
+    curl \
+    wget \
+    git \
+    && apt-get autoremove -y \
+    && apt-get clean -y \
+    && rm -rf /var/lib/apt/lists/*
+
+# Enable corepack for Yarn 4.x support
+RUN corepack enable && corepack prepare yarn@4.11.0 --activate
+
+# Set up a non-root user (node user already exists in base image)
+# Ensure the node user has proper permissions
+ARG USERNAME=node
+ARG USER_UID=1000
+ARG USER_GID=1000
+
+# Ensure node user has correct UID/GID (usually already 1000:1000 in the node image)
+RUN if [ "$USER_GID" != "$(id -g $USERNAME)" ]; then \
```

Comment:
> usually already 1000:1000 in the node image

Could you confirm this? You could build the devcontainer and verify yourself no? I don't want unnecessary code in here.